src/flock_server/keybase.py

```diff
@@ -202,11 +202,62 @@ def _usage(self, cmd):
         )
 
     async def list_users(self, bot, event, args):
-        r = Search(index="user").query("match_all").execute()
-        users = [
-            "**{}** :point_right: **{}**".format(str(hit["username"]), str(hit["name"]))
-            for hit in r
-        ]
+        # Get all users
+        user_r = Search(index="user").query("match_all").execute()
+
+        # Start gathering data on users
+        users = {}
+        for user_hit in user_r:
+            users[user_hit["username"]] = {"name": user_hit["name"]}
+
+            # Get last updated
+            try:
+                r = (
+                    Search(index="flock-*")
+                    .query("match", hostIdentifier=user_hit["username"])
+                    .sort("-@timestamp")
+                    .execute()
+                )
+                if len(r) > 0:
+                    hit = r[0]
+                    users[user_hit["username"]]["last_updated"] = hit["calendarTime"]
+            except RequestError:
+                # Ignoring this exception, because it will get triggered if an index it's searching doesn't
+                # have a mapping for @timestamp, which happens in the tests. And there doesn't seem to be
+                # a way to use elasticsearch_dsl to call `.sort` with `ignore_unmapped`...
+                # RequestError(400, 'search_phase_execution_exception', 'No mapping found for [@timestamp] in order to sort on')
+                pass
+
+            # Get OS version
+            try:
+                r = (
+                    Search(index="flock-*")
+                    .query("match", hostIdentifier=user_hit["username"])
+                    .query("match", name="os_version")
+                    .sort("-@timestamp")
+                    .execute()
+                )
+                if len(r) > 0:
+                    hit = r[0]
+                    users[user_hit["username"]]["os_version"] = f"{hit['columns']['name']} {hit['columns']['version']}"
+            except RequestError:
+                pass
+
+        # Map names to usernames
+        names = {}
+        for username in users:
+            names[users[username]["name"]] = username
```

Comment:
Come to think of it, why have a different hash at all? Why not just create a users object with the structure:

```
users = {
    "Simon Fondrie-Teitler" : {
        "username": "simonft",
        "last_updated": <time>,
        "os_version": "Debian 10",
    },
    "Micah Lee'" {
        "username": "micahflee",
        "last_updated": <time>,
        "os_version": "OSX 10.12",
    },
}
```
Then you can just do:
```
    for userame, info in sorted(users):
            response_str += f"**{info['name']}**\n"
            response_str += "".join([f"{key} :point_right: {value}\n" for key, value in info.items()])
            response_str += "\n"
```

Reply:
The reason I didn't do this is because there's no guarantee that names are unique, only that host UUIDs are unique. So if there are two people named "Micah Lee", one of them will get clobbered when building the dictionary. This is why I made the keys for `users` dictionary be the host UUID, because those are unique.

So the purpose of the separate `names` dict is just to make sure they're displayed to the user in alphabetical order by name, as opposed to ordered by host UUID, which is basically meaningless because the UUIDs are random.